### ISSUE TITLE
transformers bert 模型 onnx 格式推理正常，转为mnn 之后推理值固定为index =1 

### ISSUE THREAD

**Issue by dshwei** [alibaba/MNN#2283]
# 平台:  
Processor	12th Gen Intel(R) Core(TM) i7-12700   2.10 GHz
Installed RAM	16.0 GB (31.7 GB usable)
System type	Linux bxt301 4.14.48 #1 SMP Wed Jul 14 23:34:49 CST 2021 x86_64 x86_64 x86_64 GNU/Linux

  

# Github版本:
# Github Version:
2.4.0

    inputs = interpreter.getSessionInputAll(session)
    print("【before resize inputs 】",inputs)

    # 输入 resizeTensor
    interpreter.resizeTensor(inputs["attention_mask"], (1,128))
    interpreter.resizeTensor(inputs["token_type_ids"], (1,128))
    interpreter.resizeTensor(inputs["input_ids"], (1,128))

    # 重新计算形状分配内存
    interpreter.resizeSession(session)

    print("after resize inputs 】",inputs)

**Comment by jxt1234** [comment]
?? 报什么错呢?

**Comment by dshwei** [comment]
没有报错，推理结果一直是固定在了index =1的标签

**Comment by dshwei** [comment]
> 没有报错，推理结果一直是固定在了index =1的标签

没有报错，推理结果一直是固定在了index =1的标签

**Comment by jxt1234** [comment]
完整代码是?

**Comment by dshwei** [comment]
> 

上面的代码不完整：完整代码如下：
import MNN 
import numpy as np
from transformers import AutoTokenizer

MNN_GPU_TUNING_FAST   = 1 << 4
MNN_GPU_MEMORY_BUFFER = 1 << 6
MNN_GPU_MEMORY_IMAGE  = 1 << 7


def mode_run(model_path):

    np.random.seed(0)
    interpreter = MNN.Interpreter(model_path)
    
    interpreter.setSessionMode(8)
    interpreter.setSessionHint(0, 20)

    config = {}
    config['backend']   = "OPENCL"
    config['precision'] = "high"
    config['numThread'] = MNN_GPU_TUNING_FAST | MNN_GPU_MEMORY_BUFFER # Buffer Mode
    session = interpreter.createSession(config)

    inputs = interpreter.getSessionInputAll(session)
    print("【before resize inputs 】",inputs)

    # 输入 resizeTensor
    interpreter.resizeTensor(inputs["attention_mask"], (1,128))
    interpreter.resizeTensor(inputs["token_type_ids"], (1,128))
    interpreter.resizeTensor(inputs["input_ids"], (1,128))

    # 重新计算形状分配内存
    interpreter.resizeSession(session)

    print("after resize inputs 】",inputs)

    tokenizer = AutoTokenizer.from_pretrained("yidong_pretrain_model")
    token_inputs = tokenizer("刘德华有没有真正意义上属于自己主演的电影？", return_tensors="np",max_length=128,truncation=True,padding="max_length")
    print(token_inputs)


    attention_mask_tensor = MNN.Tensor(inputs["attention_mask"].getShape(), MNN.Halide_Type_Int,token_inputs.get("attention_mask").astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["attention_mask"].copyFrom(attention_mask_tensor)


    token_type_ids_tensor = MNN.Tensor(inputs["token_type_ids"].getShape(), MNN.Halide_Type_Int,token_inputs.get("attention_mask").astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["token_type_ids"].copyFrom(token_type_ids_tensor)


    input_ids_tensor = MNN.Tensor(inputs["input_ids"].getShape(), MNN.Halide_Type_Int,token_inputs.get("attention_mask").astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["input_ids"].copyFrom(input_ids_tensor)

    tmp_tensor = np.zeros((1,15),dtype=np.float32)
    interpreter.runSession(session)

    outputs = interpreter.getSessionOutput(session)

    print("【outputs shape】",outputs.getShape())

    out_tensor = MNN.Tensor(outputs.getShape(), MNN.Halide_Type_Float, tmp_tensor, MNN.Tensor_DimensionType_Caffe)

    outputs.copyToHostTensor(out_tensor)

    data = out_tensor.getData()
    print("【out_tensor 】",data)
    index  = np.argmax(data)
    print("【index 】",index)


if __name__ == "__main__":
    model_path = "bert_cls.mnn"
    mode_run(model_path)

**Comment by dshwei** [comment]
> 完整代码是?

onnx 模型怎么传给你

**Comment by jxt1234** [comment]
google 云盘，或者发我邮箱

**Comment by dshwei** [comment]
> 或者发我邮箱

邮箱吧
你的邮箱发我一下

**Comment by jxt1234** [comment]
[jxt1234@zju.edu.cn](mailto:jxt1234@zju.edu.cn)

**Comment by dshwei** [comment]
> [jxt1234@zju.edu.cn](mailto:jxt1234@zju.edu.cn)

已发送

**Comment by jxt1234** [comment]
你调用 onnxruntime 推理 onnx 模型的代码也发一下吧

**Comment by jxt1234** [comment]
另外，把 onnx 模型发我邮箱，mnn 文件发我没什么用

**Comment by dshwei** [comment]
> 
两个onnx 文件都发给你了，一个固定shape输入，一个是动态shape 输入

**Comment by jxt1234** [comment]
调用 onnxruntime 推理 onnx 模型的代码呢？需要确保下 onnx 本身是可用的

**Comment by jxt1234** [comment]
demo 里面代码这一句执行不了，没有 bert_pretrain_model 这个目录
AutoTokenizer.from_pretrained("bert_pretrain_model")

**Comment by Goswell** [comment]
@dshwei 你代码里的token_type_ids_tensor和input_ids_tensor好像写错了，按现在的写法这两个值都是token_inputs["attention_mask"]？

**Comment by dshwei** [comment]
> 
问题已解决，是这里的问题，谢谢，
